### PR TITLE
Forgot password link displays new screen before reset password code

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

This PR has the following changes
- Change text from 'Forgot Password' to 'Forgot Password?' in <SignIn />
- When user clicks on the link show a screen with the following options:
  - Add Reset Password Option
  - Social Connection and Web3 Options
  - If other authentication factors are turned on
     - Email Code
     - Email Link
     - SMS Code

https://github.com/clerkinc/javascript/assets/19269911/a9d55752-150a-4127-b305-46858067a844

### CODE REVIEW

File: packages/localizations/src/en-US.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const enUS: LocalizationResource = {
   formFieldInputPlaceholder__organizationSlug: '',
   formFieldError__notMatchingPasswords: `Passwords don't match.`,
   formFieldError__matchingPasswords: 'Passwords match.',
-  formFieldAction__forgotPassword: 'Forgot password',
+  formFieldAction__forgotPassword: 'Forgot password?',
```
Comment (desiprisg) [inline packages/localizations/src/en-US.ts:55]: 🙃 can we add the question mark to the rest of the localization packages as well?

File: packages/localizations/src/en-US.ts [hunk 2]
```diff
@@ -138,6 +138,11 @@ export const enUS: LocalizationResource = {
       subtitle: 'to continue to {{applicationName}}',
       actionLink: 'Use another method',
     },
+    forgotPasswordAlternativeMethods: {
+      title: 'Forgot Password?',
+      alternativeMethods: 'Or, sign in with another method.',
```
[the remark this answers is not in the document]
Comment (desiprisg) [reply packages/localizations/src/en-US.ts:143]: Actually, it's more like a subtitle, right?

File: packages/clerk-js/src/ui/components/SignIn/ForgotPasswordAlternativeMethods.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const resetPasswordFactor = useResetPasswordFactor();
+
+  const { firstPartyFactors, hasAnyStrategy } = useAlternativeStrategies({
+    filterOutFactor: props.currentFactor,
+  });
+
+  if (!resetPasswordFactor) {
+    // Make TS happy, we are handling this logic further up in the tree already,
+    // so we will never return null here.
+    return null;
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/SignIn/ForgotPasswordAlternativeMethods.tsx:30]: How is this already handled?

File: packages/localizations/src/en-US.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (panteliselef) [reply packages/localizations/src/en-US.ts:143]: it's just a label, no action, no subtitle. I could replace it with `moreMethods__label` ?

[the remark this answers is not in the document]
Comment (desiprisg) [reply packages/localizations/src/en-US.ts:143]: Maybe `divider` although we don't use that anywhere else? @nikosdouvlis any thoughts?

**Review by desiprisg — APPROVED** [review]

**Comment by desiprisg** [comment]
Let's test this on staging

**Pull request merged**